cmd/collector/app/metrics.go

```diff
@@ -80,15 +81,16 @@ func NewSpanProcessorMetrics(serviceMetrics metrics.Factory, hostMetrics metrics
 		spanCounts[otherFormatType] = newCountsBySpanType(serviceMetrics.Namespace("", map[string]string{"format": otherFormatType}))
 	}
 	m := &SpanProcessorMetrics{
-		SaveLatency:    hostMetrics.Timer("save-latency", nil),
-		InQueueLatency: hostMetrics.Timer("in-queue-latency", nil),
-		SpansDropped:   hostMetrics.Counter("spans.dropped", nil),
-		BatchSize:      hostMetrics.Gauge("batch-size", nil),
-		QueueLength:    hostMetrics.Gauge("queue-length", nil),
-		ErrorBusy:      hostMetrics.Counter("error.busy", nil),
-		SavedBySvc:     newMetricsBySvc(serviceMetrics, "saved-by-svc"),
-		spanCounts:     spanCounts,
-		serviceNames:   hostMetrics.Gauge("spans.serviceNames", nil),
+		SaveLatency:     hostMetrics.Timer("save-latency", nil),
+		InQueueLatency:  hostMetrics.Timer("in-queue-latency", nil),
+		SpansDropped:    hostMetrics.Counter("spans.dropped", nil),
+		BatchSize:       hostMetrics.Gauge("batch-size", nil),
+		QueueLength:     hostMetrics.Gauge("queue-length", nil),
+		ErrorBusy:       hostMetrics.Counter("error.busy", nil),
+		SavedBySvc:      newMetricsBySvc(serviceMetrics, "saved-by-svc"),
```

Comment:
I'd prefer if we used tags instead: metric name = saved-by-svc, tag = success|fail

Reply:
although, technically it'll be a breaking change but we've gone ahead with metric changes without considering it a breaking change

Reply:
If we want to take this approach, then there are some other places that should be changed also - e.g. `jaeger_find_traces_errLatency_count` and `jaeger_find_traces_okLatency_count`. But I agree it would be better to have it as a tag on the metric.